lib/virtual-fs/src/union_fs.rs

```diff
@@ -178,6 +180,43 @@ impl UnionFileSystem {
             }
         }
 
+        // also take into account partial matchings
```

Comment:
What is partial matchings? This seems to be doing things that the previous for loop should handle